### ISSUE TITLE
Reader skips chapters (when downloaded)

### ISSUE THREAD

**Issue by antocs** [tachiyomiorg/tachiyomi#7472]
### Steps to reproduce

Download 2 chapters, start reading the first one then swipe to change page.

### Expected behavior

It should work normally, continuing reading the actual chapter.

### Actual behavior

It suddenly skips to the 2nd chapter.

### Tachiyomi version

Preview r4553

### Android version

Android 12

### Device

Samsung Galaxy S10

### Other details

(All this doesn't happen if chapters are not downloaded, deleting and downloading again doesn't solve it)
As shown in the videos, the moment I change page I skips to the next chapter directly.

https://user-images.githubusercontent.com/22413659/177612800-367ea418-d937-48eb-8fe4-85e13be8b77c.mp4

https://user-images.githubusercontent.com/22413659/177612811-effcf3a1-ea9f-46ac-b2cb-1285d331c907.mp4

In this last video I used the slider without reaching the last page, see the chapter number on top skipping like crazy.
https://streamable.com/kojvg4

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.4](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by antocs** [comment]
I made another test: It only happens if the next unread chapter is downloaded.
If I have chapters 1 2 and 3, but 2 is not downloaded, I can read normally chapter 1.

**Comment by yohanim** [comment]
I'd say it's related to #7391 even though worst on your side

**Comment by antocs** [comment]
I had problems with webtoon reading mode too, but I have had many tests for that.

**Comment by antocs** [comment]
another test: Setting reading mode left to right (so western-style and not manga) doesn't trigger the bug.

**Comment by hugocornago** [comment]
Similar issue, when reading, in webtoon format, two consecutive chapters (both downloaded), sometimes the second one starts on the last page. I suppose it may be not be resetting the page number correctly when changing chapters.

**Comment by SkillGG** [comment]
This bug sometimes causes me to not be able to read the chapter to the end if it's the last one. It tries to skip to next chapter. But there is none.
And going backwards causes the "forward skip" too

**Comment by antocs** [comment]
Today I started a new manga I downloaded a few days ago, I started it normally with default reading mode, I forgot about the issue, but then I noticed that the bugged didn't happen.
So I went and tried downloading and reading the 2nd manga in the first post (2nd and 3rd video on streamable) and the bug didn't happen.
I guess something fucked up during the download and the cache would keep it bugged until I closed the app (I have automatic clear-cache on closing app).
